contracts/data-storage/andromeda-form/src/execute.rs

```diff
@@ -0,0 +1,418 @@
+                            &(new_id.u64(), sender.clone()),
+                            &SubmissionInfo {
+                                submission_id: new_id.u64(),
+                                wallet_address: sender.clone(),
+                                data,
+                            },
+                        )?;
+                        SUBMISSION_ID.save(ctx.deps.storage, &new_id)?;
+                    } else {
+                        return Err(ContractError::CustomError {
+                            msg: "Multiple submissions are not allowed".to_string(),
+                        });
+                    }
+                }
+            }
+        }
+        ValidateDataResponse::Invalid { msg } => return Err(ContractError::CustomError { msg }),
+    }
+
+    let response = Response::new()
+        .add_attribute("method", "submit_form")
+        .add_attribute("submission_id", new_id)
+        .add_attribute("sender", sender.clone());
+
+    Ok(response)
+}
+
+pub fn execute_delete_submission(
+    ctx: ExecuteContext,
+    submission_id: u64,
+    wallet_address: AndrAddr,
+) -> Result<Response, ContractError> {
+    nonpayable(&ctx.info)?;
+    let sender = ctx.info.sender;
+    ensure!(
+        ADOContract::default().is_owner_or_operator(ctx.deps.storage, sender.as_ref())?,
+        ContractError::Unauthorized {}
+    );
+
+    let address = wallet_address.get_raw_address(&ctx.deps.as_ref())?;
+    submissions()
+        .load(ctx.deps.storage, &(submission_id, address.clone()))
+        .map_err(|_| ContractError::CustomError {
+            msg: format!(
+                "Submission does not exist - Submission_id {:?}, Wallet_address {:?}",
+                submission_id, wallet_address
+            ),
+        })?;
+    submissions().remove(ctx.deps.storage, &(submission_id, address.clone()))?;
+
+    let response = Response::new()
+        .add_attribute("method", "delete_submission")
+        .add_attribute("submission_id", Uint64::from(submission_id))
+        .add_attribute("sender", sender);
+
+    Ok(response)
+}
+
+pub fn execute_edit_submission(
+    ctx: ExecuteContext,
+    submission_id: u64,
+    wallet_address: AndrAddr,
+    data: String,
+) -> Result<Response, ContractError> {
+    nonpayable(&ctx.info)?;
+    let sender = ctx.info.sender;
+
+    let config = CONFIG.load(ctx.deps.storage)?;
+    let allow_edit_submission = config.allow_edit_submission;
+    ensure!(
+        allow_edit_submission,
+        ContractError::CustomError {
+            msg: "Edit submission is not allowed".to_string(),
+        }
+    );
+    let (expiration, _) = get_and_validate_start_time(&ctx.env, None)?;
+    let current_time = milliseconds_from_expiration(expiration)?;
+    let saved_start_time = config.start_time;
+    let saved_end_time = config.end_time;
+    // validate if the Form is opened
+    validate_form_is_opened(current_time, saved_start_time, saved_end_time)?;
```

Comment:
Same here, maybe let this function just take `env` and `config` as variables and let it handle the rest within it